fix: Allow JPEG and GIF images as profile photos #2332

islathehut opened the pull request:
* Fixes `UserProfileStore` and `UserProfileContextMenu` to allow JPEG and GIF profile photos as well as PNG.
* Updates E2E tests to include JPEG and GIF cases on profile photo tests
* Adds new backend unit tests for img validation
* Adds new `npm` scripts at the root for conveniently running the desktop app locally instead of having to `cd` into the desktop package folder and linting all packages at once.
* Adds new `npm` scripts for local binary compilation and updates E2E tests allow using local binaries
* Adds documentation for new local binary compilation and testing scripts
* Fixes issues I had while running tor tests locally
* Fix the issue with `ContextMenuProps` having type `unknown` for the field `visible`

GitHub Issue: https://github.com/TryQuiet/quiet/issues/2332


### Pull Request Checklist

- [x] I have linked this PR to a related GitHub issue.
- [x] I have added a description of the change (and Github issue number, if any) to the root [CHANGELOG.md](https://github.com/TryQuiet/quiet/blob/develop/CHANGELOG.md).

### (Optional) Mobile checklist

Please ensure you completed the following checks if you did any changes to the mobile package:

- [ ] I have run e2e tests for mobile
- [ ] I have updated base screenshots for visual regression tests

Commit:
Allow JPEG and GIF images as profile photos

- update desktop component to allow jpeg and gif
- update userprofilestore to support jpeg and gif
- add start:desktop script at root for ease

islathehut commented:
Fixes #2332

Commit:
Update CHANGELOG files

Commit:
Accidentally included bad string

Commit:
Add root-level lint script and fix linting errors

Commit:
Update E2E tests to include JPEG and GIF profile photo tests

leblowl reviewed (approved):
Thanks. I would move the changelog changes under unreleased, but I can also do it later. Here are some of my thoughts around validation for this feature: https://github.com/TryQuiet/quiet/discussions/2173 in case you are interested ... perhaps the image magic byte sequence check is unnecessary, but it doesn't seem like a big deal.

Commit:
Update formatting of CHANGELOG files

islathehut commented:
> Thanks. I would move the changelog changes under unreleased, but I can also do it later. Here are some of my thoughts around validation for this feature: #2173 in case you are interested ... perhaps the image magic byte sequence check is unnecessary, but it doesn't seem like a big deal.

Updated the CHANGELOG files with that formatting.  I think for now we can leave the magic byte checks but would be curious if we could do away with it while still maintaining assurance that images are what they say they are.  Part of me also wonders if we should merge the type check (e.g. `data:image/png;base64`) with the magic byte check for performance (since we'll only have to run one header check per image) and some extra assurance (that way we are validating that not only is the type correct on the uploaded image but it matches the magic byte).

Commit:
Refactor profile photo validation

islathehut commented:
Refactored the photo validation so its consolidated and connected the file type check with the magic number check for added assurance/performance.

Commit:
Add comments

Commit:
Fix linting (thought I ran this already)

EmiM commented:
Could you add backend unit tests for new file types?

EmiM commented on packages/e2e-tests/src/tests/userProfile.test.ts:
```diff
@@ -106,11 +106,25 @@ describe('User Profile Feature', () => {
+    await menu.uploadJPEGPhoto()
+  })
+
+  it('Owner updates their profile photo with GIF', async () => {
```
Could you check locally if userProfile test is passing?
* Last of "Owner updates their profile photo with..." tests uploads gif photo so if the upload suppose to be successful then assertion in "Owner's message with profile photo is visible on channel" should be updated from checking png to gif
* Could you maybe add assertion at the end of each of "Owner updates their profile photo with..." cases to make sure that e.g users do not see upload error message?

islathehut replied:
Added assertions and testing locally now

islathehut committed:
Update e2e tests and add ability to run tests locally against locally generated dmg

islathehut committed:
Add backend tests for gif and jpeg validation

islathehut committed:
Separate test script for local binaries

islathehut committed:
Update naming convention and mild refactor

islathehut committed:
Add documentation on running tests against local binaries

islathehut committed:
Add unit tests to backend for validatePhoto and mild refactor

islathehut committed:
Fix tor test issue I was seeing locally, add tor logging and skip running kills when no process exists

Tor was only spawning one process when running on my machine (might be an arm64 thing) so this test was failing since it expected two processes to close but there was just one

islathehut committed:
Linting

islathehut committed:
Merge branch 'develop' into bug/2332-allow-jpeg-profile-photos

islathehut committed:
Fix tests hanging on getting back button

islathehut committed:
Linting

islathehut committed:
Use unique data-testids for back arrow (thank you Emilia)

islathehut committed:
Linting
